### PR TITLE
Pt lima bidding

### PR DESCRIPTION
**Pull request by ThomasArts**
Bidding seems to work, need more tests though

### CODE REVIEW

File: apps/aeprimop/src/aeprimop.erl [hunk 1]
```diff
@@ -1647,6 +1698,25 @@ assert_not_name(NameHash, S) ->
...
+    case find_name_auction(AuctionHash, S) of
+        {_, _} -> runtime_error(name_already_in_auction);
+        none   -> ok
+    end.
+
+assert_valid_overbid(Protocol, NameAscii, NewNameFee, OldNameFee) ->
+    MinNameFee = aec_governance:name_claim_fee(NameAscii, Protocol),
+    Progression = aec_governance:name_claim_bid_increment(),
+    %% Stay within integer computations New bid must be Progression % higher
+    case (NewNameFee - OldNameFee) * 100 >= MinNameFee * Progression of
```
Comment (sennui) [inline apps/aeprimop/src/aeprimop.erl]: I think we should not include MinNameFee here. For popular domains, spam incentive will be the highest and MinNameFee the lowest in comparison to {New,Old}NameFee.

Comment (ThomasArts) [reply apps/aeprimop/src/aeprimop.erl]: Indeed, now we compare to OldNameFee instead

File: apps/aens/src/aens_auctions.erl [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+-define(AUCTION_TYPE, name_auction).
+-define(AUCTION_VSN, 1).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec new(aens_hash:auction_hash(), aec_keys:pubkey(), non_neg_integer(), non_neg_integer(), aec_blocks:height()) -> auction().
+new(AuctionHash, BidderId, NameFee, DeltaTTL, BlockHeight) ->
```
Comment (hanssv) [inline apps/aens/src/aens_auctions.erl]:
```suggestion
new(AuctionHash, Bidder, NameFee, DeltaTTL, BlockHeight) ->
```

File: apps/aens/src/aens_auctions.erl [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    NameHash = aens_hash:from_auction_hash(AuctionHash),
+    #auction{id        = aeser_id:create(name, NameHash),
+             bidder_id = aeser_id:create(account, BidderId),
```
Comment (hanssv) [inline apps/aens/src/aens_auctions.erl]:
```suggestion
             bidder_id = aeser_id:create(account, Bidder),
```

File: apps/aeprimop/src/aeprimop.erl [hunk 4]
```diff
@@ -725,30 +723,69 @@ name_preclaim({AccountPubkey, CommitmentHash, DeltaTTL}, S) ->
...
+            {Auction, S1} = get_name_auction(AuctionHash, name_not_in_auction, S),
+            PreviousBidderPubkey = aens_auctions:bidder_pubkey(Auction),
+            PreviousBid = aens_auctions:name_fee(Auction),
+            assert_name_bid_fee(NameAscii, NameFee, S#state.height), %% just in case, consensus may have changed
+            assert_valid_overbid(Protocol, NameAscii, NameFee, aens_auctions:name_fee(Auction)),
+            NewAuction = aens_auctions:new(AuctionHash, AccountPubkey, NameFee, Timeout, S1#state.height),
+            %% Return the tokens hold in the previous bid
+            {PreviousBidderAccount, S2} = get_account(PreviousBidderPubkey, S1),
+            S3 = account_earn(PreviousBidderAccount, PreviousBid, S2),
+            S4 = delete_x(name_auction, AuctionHash, S3),
```
Comment (hanssv) [inline apps/aeprimop/src/aeprimop.erl]: unnecessary to delete when putting the same key?

**Pull request merged**